### PR TITLE
docs: add `gcp-auth` to third-party plugin list

### PR DESCRIPTION
**Pull request by AndyClausen**
**What's the problem this PR addresses?**

This will add a [third-party plugin](https://github.com/AndyClausen/yarn-plugin-gcp-auth)  to the plugin list. The plugin will add a GCP access token to the NPM authentication header to any requests going to Google Artifact Registry repos - much like the existing [az-cli-auth](https://github.com/FishandRichardsonPC/yarn-plugin-az-cli-auth).

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by AndyClausen** [comment]
I'm not sure why that one integration test failed, but I _highly_ doubt it's because of this edit

File: packages/gatsby/content/features/plugins.md [hunk 1]
```diff
@@ -71,6 +71,8 @@ This is just a centralized list of third-party plugins to make discovery easier.
 
 - [**az-cli-auth**](https://github.com/FishandRichardsonPC/yarn-plugin-az-cli-auth) by [**Fish & Richardson P.C**](https://fr.com) - Uses the az cli to generate auth tokens when using azure devops repos
 
+- [**gcp-auth**](https://github.com/AndyClausen/yarn-plugin-gcp-auth) by [**Andreas Clausen**](https://github.com/AndyClausen) - Uses gcloud to generate access tokens when using Google Artifact Regitry repos
```
Comment (paul-soporan) [inline packages/gatsby/content/features/plugins.md:74]:
```suggestion
- [**gcp-auth**](https://github.com/AndyClausen/yarn-plugin-gcp-auth) by [**Andreas Clausen**](https://github.com/AndyClausen) - Uses gcloud to generate access tokens when using Google Artifact Registry repos
```

**Review by paul-soporan — APPROVED** [review]
> I'm not sure why that one integration test failed, but I highly doubt it's because of this edit

That test is a bit flaky so you can just ignore it, I restarted the workflow to get it to pass.

**Comment by AndyClausen** [comment]
Ping - needs restart again

**Comment by merceyz** [comment]
It's unlikely a restart will let it pass, we'll have to wait for @arcanis to merge it

**Pull request merged**